js/card.js

```diff
@@ -0,0 +1,99 @@
+        return forms[1];
+      default:
+        return forms[2];
+    }
+  };
+
+  var roomsFlexNormalize = function (number) {
+    var forms = ['комнат', 'комната', 'комнаты'];
+    return flexNormalize(number, forms);
+  };
+
+  var guestsFlexNormalize = function (number) {
+    var forms = ['гостей', 'гостя', 'гостей'];
+    return flexNormalize(number, forms);
+  };
+
+  var createFeature = function (feature) {
+    var featureElement = document.createElement('li');
+    featureElement.className = 'popup__feature  popup__feature--' + feature;
+
+    return featureElement;
+  };
+
+  var createPhoto = function (photo) {
+    var photoElement = document.createElement('img');
+    photoElement.src = photo;
+    photoElement.className = 'popup__photo';
+    photoElement.style = 'width: 45px; height: 40px;';
+    photoElement.alt = 'Фотография жилья';
+
+    return photoElement;
+  };
+
+  var appendElement = function (data, container, fn) {
+    var fragment = document.createDocumentFragment();
+
+    container.innerHTML = '';
+    data.forEach(function (element) {
+      fragment.appendChild(fn(element));
+    });
+    container.appendChild(fragment);
+  };
+
+
+  var renderCard = function (card) {
+    var cardTemplate = document.querySelector('#card').content.querySelector('.map__card');
+    var cardElement = cardTemplate.cloneNode(true);
+    var featuresList = cardElement.querySelector('.popup__features');
+    var photoList = cardElement.querySelector('.popup__photos');
+    var capacityContent = (card.offer.rooms || 1) + ' ' + roomsFlexNormalize(card.offer.rooms || 1) + ' для ' + (card.offer.guests || 1) + ' ' + guestsFlexNormalize(card.offer.guests || 1);
+    featuresList.innerHTML = '';
+    photoList.innerHTML = '';
+
+    cardElement.querySelector('.popup__avatar').src = card.author.avatar;
+    cardElement.querySelector('.popup__title').textContent = card.offer.title;
+    cardElement.querySelector('.popup__text--address').textContent = card.offer.address;
+    cardElement.querySelector('.popup__text--price').innerHTML = card.offer.price + '&#x20bd;<span>/ночь</span>';
+    cardElement.querySelector('.popup__type').textContent = houseTypes[card.offer.type];
+    cardElement.querySelector('.popup__text--capacity').textContent = capacityContent;
+    cardElement.querySelector('.popup__text--time').textContent = 'Заезд после ' + card.offer.checkin + ', выезд до ' + card.offer.checkout;
+    cardElement.querySelector('.popup__description').textContent = card.offer.description;
+    appendElement(card.offer.features, featuresList, createFeature);
+    appendElement(card.offer.photos, photoList, createPhoto);
+
+    return cardElement;
+  };
+
+  var makeCardElements = function (cardElement) {
+    var mapFilters = document.querySelector('.map__filters-container');
+    var currentLocation = cardElement;
+
+    mapFilters.before(renderCard(currentLocation));
+  };
+
+
+  window.card = {
+    makeCardElements: makeCardElements
+  };
```

Comment:
В имени свойства не должно содержаться имя объекта, можно переименовать так
```suggestion
  window.card = {
    makeElements: makeCardElements
  };
```